File: tests/files/extract_var_works_before.java
package org.example;

public class App {

    public static void main(String[] args) {

        System.out.println("foo");

        while (true) {
            System.out.println("foo");
            break;
        }

        do {
            System.out.println("foo");
        } while (false);

        if (true) {
            System.out.println("foo");
        } else {
            System.out.println("foo");
        }

        for (int i = 0; i < 5; i++) {
            System.out.println("foo");
        }

    }
}

File: tests/files/extract_var_works_before.lua
local function bar()
  print("foo")
end
print("foo")

do
  print("foo")
end

while false do
  print("foo")
end

repeat
  print("foo")
until true

if true then
  print("foo")
else
  print("foo")
end

for i = 1, 2 do
  print("foo")
end
local baz = function()
  print("foo")
end
